### PR TITLE
refactor: improve consistency of extras

### LINKED ISSUE #31: refactor: improve consistency of extras
### Is your feature request related to a problem? Please describe
The `extras` field of `TimeStep` can contain environment information useful for decision-making (e.g. `Connect4`'s current player ID) or environment metrics (e.g. BinPack's volume utilisation). There is an inconsistency in what the `extras` field is used for as it is sometimes meant to be used by the algorithm and sometimes just logged as a metric.

### Describe the solution you'd like
We should move any algorithm-related information from extras to the environment observation (e.g. Connect4's observation could have another field called `current_player` or something). We should update the documentation/docstrings accordingly to explicitly mention that `TimeStep.extras` does not contain stuff that is meant to be observed by the agent as those should be in the observation.

---
### TODOs
- [ ] adapt docstrings, doc, codes, etc to make explicit the fact that `TimeStep.extras` does not contain any info meant to be observed
- [ ] move agent-specific extras (e.g. Connect4's current player ID) to environment observations

### PR DESCRIPTION
**Pull request by dluo96**
Closes #31.

- [ ]  Adapted docstrings around `extras` to make it clear that `TimeStep.extras` does **not** contain any information meant to be observed by the agent. Only `observation` does. 
- [ ] Move agent-specific quantities (for example the `current_player` in the Connect4 environment's `extras` field) from `extras` to the `observation` field.

### CODE REVIEW

File: jumanji/environments/games/connect4/env.py [hunk 1]
```diff
@@ -141,21 +143,23 @@ def step(self, state: State, action: Action) -> Tuple[State, TimeStep[Observatio
         # creating next state
         next_state = State(current_player=jnp.int8(next_player), board=new_board)
 
-        obs = Observation(board=new_board, action_mask=action_mask)
+        obs = Observation(
+            board=new_board,
+            action_mask=action_mask,
+            current_player=jnp.int8(next_player),
```
Comment (clement-bonnet) [inline jumanji/environments/games/connect4/env.py]: To make this a little cleaner, we could cast next_player to `jnp.int8` directly when it's created at line 132: `next_player = jnp.int8((state.current_player + 1) % self.n_players)` and use this at line 144 as well.

File: jumanji/types.py [hunk 2]
```diff
@@ -72,8 +72,9 @@ class TimeStep(Generic[Observation]):
             also valid in place of a scalar array.
         extras: environment metrics or things to be seen by the agent but not directly
```
Comment (clement-bonnet) [inline jumanji/types.py]: This becomes slightly confusing with the end of the paragraph. Maybe I would change it to something along the lines: "extras: environment metrics or custom data returned by the environment, not observed by the agent".

File: jumanji/environments/games/connect4/env.py (hunk 1 above)
Comment (dluo96) [reply jumanji/environments/games/connect4/env.py]: Good idea, have done

File: jumanji/types.py (hunk 2 above)
Comment (dluo96) [reply jumanji/types.py]: Have updated the description 👍
